### PR TITLE
Update default etcd server to 3.4.9

### LINKED ISSUE #91266: Update to etcd 3.4.9 in Kubernetes 1.19
Bump etcd version to 3.4.9 to pick up a few bug fixes, notably: 
 - Fix inconsistency between WAL and server snapshot. (https://github.com/etcd-io/etcd/pull/11888)
 - Fix a data corruption bug when etcd auth is enabled. (https://github.com/etcd-io/etcd/pull/11652)

Todo:
- [x] Update etcd client version (#92075)
- [x] Update etcd server version in Makefile under `cluster/images/etcd` (#92078)
- [x] Test and promote images from staging to prod repo (https://github.com/kubernetes/k8s.io/pull/973)
- [ ] Update default etcd server version everywhere in codebase (#92349)

/sig api-machinery
/kind feature

### PR DESCRIPTION
**Pull request by jingyih**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
We updated etcd client version to 3.4.9 in #92075. We also updated Makefile for building etcd 3.4.9 images in #92078, which triggers automatic build [1]. This PR updates the default etcd server version to 3.4.9.

[1] https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/post-kubernetes-push-image-etcd/1273799163636617217

**Which issue(s) this PR fixes**:
Fixes #91266

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Update default etcd server version to 3.4.9
```

/sig api-machinery
/area etcd

### CODE REVIEW

**Comment by jingyih** [comment]
cc @BenTheElder @dims @wenjiaswe @kubernetes/release-engineering

**Comment by saschagrunert** [comment]
Hey, one question: with respect to the release note, what is the _user facing_ change? Is kubeadm using the new etcd version?

Internal test and build environments do not count as “user facing”. 🙂

**Comment by BenTheElder** [comment]
/lgtm
/approve

> Hey, one question: with respect to the release note, what is the user facing change? Is kubeadm using the new etcd version?

Everything in the repo is included in the release, regardless.

**Comment by justaugustus** [comment]
/hold cancel
(Promotion successful in https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/post-k8sio-cip/1274443174978260992.)

/lgtm
/approve
/retest

**Comment by BenTheElder** [comment]
kubeadm is still trying to pull from `k8s.gcr.io` (`k8s.gcr.io/etcd:3.4.9-0`)

on monday k8s.gcr.io is supposed to be live from k8s infra though ...

**Comment by justaugustus** [comment]
> kubeadm is still trying to pull from `k8s.gcr.io` (`k8s.gcr.io/etcd:3.4.9-0`)
> 
> on monday k8s.gcr.io is supposed to be live from k8s infra though ...

@BenTheElder -- Heh. I was just about to ping you.
I'd say we wait on this PR until the vanity domain flip starts.

**Review by neolit123 — COMMENTED** [review]
/approve

**Comment by liggitt** [comment]
the 1.17 aggregator e2e failure looks related:

> ContainersNotReady containers with unready status: [etcd]

File: test/utils/image/manifest.go
Comment (liggitt) [inline test/utils/image/manifest.go:217]: does this image exist?

Comment (liggitt) [reply test/utils/image/manifest.go:217]: this is pointing at `k8s.gcr.io`... does the 3.4.9 image exist there?

**Comment by liggitt** [comment]
doesn't look like the e2e or kind image is available:

docker pull k8s.gcr.io/etcd:3.4.9
Error response from daemon: manifest for k8s.gcr.io/etcd:3.4.9 not found: manifest unknown: Failed to fetch "3.4.9" from request "/v2/etcd/manifests/3.4.9".

docker pull k8s.gcr.io/etcd:3.4.9-0
Error response from daemon: manifest for k8s.gcr.io/etcd:3.4.9-0 not found: manifest unknown: Failed to fetch "3.4.9-0" from request "/v2/etcd/manifests/3.4.9-0".

**Comment by BenTheElder** [comment]
the k8s.gcr.io flip didn't succeed, so these images aren't reachable there.
the image is at a different registry (that will back this one when the flip
succeeds)

**Comment by liggitt** [comment]
> the image is at a different registry (that will back this one when the flip succeeds)

ok... do you know if we pursue pushing the 3.4.9 image to the k8s.gcr.io registry or wait for the registry flip?

**Comment by justaugustus** [comment]
> > the image is at a different registry (that will back this one when the flip succeeds)
> 
> ok... do you know if we pursue pushing the 3.4.9 image to the k8s.gcr.io registry or wait for the registry flip?

@liggitt @BenTheElder -- we should switch the etcd image to point at the same K8s Infra location (`us.gcr.io/k8s-artifacts-prod`) as the eventual release images (ref: https://github.com/kubernetes/k8s.io/pull/889).

```json
$ skopeo inspect docker://us.gcr.io/k8s-artifacts-prod/etcd:3.4.9-0 --raw
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1372,
         "digest": "sha256:1bf63e04727684d902d13c29827c43865b38dcae23f8e740abd6d27a8e1c64ce",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1372,
         "digest": "sha256:fab0598bcd89f13d52ca65576c4cd14e9fe83ddf4f0cddcf248df48a0f699fcb",
         "platform": {
            "architecture": "arm",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1372,
         "digest": "sha256:203609377848b73ed499f4133f1af89a22359b188ffe4928f265091ae52629ed",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1373,
         "digest": "sha256:785aba7054cd7dd62785e03c1643d4cbea8c4509907e4f02e6fa42dca073e40e",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1372,
         "digest": "sha256:404197720b8d6460243c86bfb546d18ed848e3815ab61a5db8485f0d3cb5c8c4",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      }
   ]
}
```

the next k8s.gcr.io cutover attempt is at least two weeks out, so we should probably switch to the correct registry ahead of that.

**Comment by liggitt** [comment]
ok. @jingyih ^

**Comment by BenTheElder** [comment]
I'm a bit hesitant to depend on us.gcr.io for a "prod" image ...?

File: test/utils/image/manifest.go
Comment (justaugustus) [reply test/utils/image/manifest.go:217]:
```suggestion
	configs[Etcd] = Config{prodRegistry, "etcd", "3.4.9"}
```

**Review by justaugustus — CHANGES_REQUESTED** [review]
@jingyih -- In https://github.com/kubernetes/kubernetes/blob/a463b25c9d170b14b5c3183c8604ebb906a97509/test/utils/image/manifest.go#L69-L85 you'll want to add something like:

```go
  ProdRegistry: "us.gcr.io/k8s-artifacts-prod"
```

and then add that to the `configs` map.

**Comment by liggitt** [comment]
> I'm a bit hesitant to depend on us.gcr.io for a "prod" image ...?

I agree. Looks like it's already crept in in places:

>     image: us.gcr.io/k8s-artifacts-prod/kas-network-proxy/proxy-server:v0.0.9

>     "image": "{{ pillar.get('etcd_docker_repository', 'us.gcr.io/k8s-artifacts-prod/etcd') }}:{{ pillar.get('etcd_docker_tag', '3.4.7-2') }}",

**Comment by justaugustus** [comment]
@liggitt -- the plan was to search/replace these references and merge once the VDF was complete.
Without making these changes, a variety of container image upgrades ([debian base images as an example](https://github.com/kubernetes/kubernetes/issues/58012)) would have been blocked on VDF, including some security-related ones (ref: https://github.com/kubernetes/kubernetes/pull/91387).

While hiccups come with the territory in big shifts like this, VDF is almost three months past the initial target date, and a lot of PRs would've been blocked in the meantime w/o choosing the regional target.

**Comment by liggitt** [comment]
build/test-time dependencies are really different than prod runtime dependencies... pointing the former at us.gcr.io/k8s-artifacts-prod has failure modes that are annoying, but wouldn't cause outages

**Comment by jingyih** [comment]
@justaugustus I have applied the changes you suggested. There are still some tests failed due to failing to pull from `k8s.gcr.io`. Looks like they are related to image building for node?

https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/92349/pull-kubernetes-conformance-kind-ga-only-parallel/1275817356920098817#1:build-log.txt%3A186

File: test/utils/image/manifest.go
Comment (wojtek-t) [inline test/utils/image/manifest.go:220]: IN order to unblock this change, can we promote it to the old repo in the meantime and keep using that for a bit *until the flip will really happen"?

Comment (jingyih) [reply test/utils/image/manifest.go:220]: @dims @justaugustus If we want to use the old prod repo, is there a way to promote image from new staging repo to old prod repo? Or do we have to manually push to old staging repo first?

Comment (wojtek-t) [reply test/utils/image/manifest.go:220]: From what I know the latter only (you need to push to the old staging repo)

**Comment by dims** [comment]
@jingyih please ping @listx to figure out logistics (old vs new repo)

**Comment by jingyih** [comment]
Pushed the images to the old prod registry as well.

**Comment by neolit123** [comment]
today is feature freeze for 1.19. are we not merging this change?

File: cluster/gce/manifests/etcd.manifest
Comment (BenTheElder) [inline cluster/gce/manifests/etcd.manifest:18]: this needs to switch back to the old prod registry?

**Comment by BenTheElder** [comment]
@neolit123 there's a minor bug in the PR

File: cluster/gce/manifests/etcd.manifest
Comment (jingyih) [reply cluster/gce/manifests/etcd.manifest:18]: Thanks. But both new and the old prod registry have image for this version.

**Comment by jingyih** [comment]
> today is code freeze for 1.19. are we not merging this change?

I think this PR is clean up. (https://github.com/kubernetes/kubernetes/pull/92075 is considered feature)

**Comment by jingyih** [comment]
All tests passed. Could we get this in? @neolit123 @BenTheElder @justaugustus

**Pull request merged**
